### ISSUE TITLE
Keep BasicAuthenticator for backward compatibility

### ISSUE THREAD

**Issue by mtrakal** [couchbase/couchbase-lite-android#920]
Hi,

trying to test latest build:
```
compile 'com.couchbase.lite:couchbase-lite-android:0.0.0-766'
compile 'com.couchbase.lite:couchbase-lite-android-forestdb:0.0.0-766'
```

but I get error about missing:
```
import com.couchbase.lite.android.AndroidContext;
import com.couchbase.lite.auth.BasicAuthenticator;
```

It's from: `0.0.0-750`. Build with `0.0.0-748` is ok, but newer failed.

**Comment by mtrakal** [comment]
I look into source codes, and see replace `BasicAuthenticator `with `PasswordAuthorizer`, so I'm abble to resolve `PasswordAuthorizer`.

But still have a problem with: `AndroidContext`

**Comment by mtrakal** [comment]
After 3rd resync gradle, now resolve resources ok :/. Have no idea where was a problem...

**Comment by hideki** [comment]
Hi @mtrakal,
Sorry for changing class name without notifying community. I have tried to make consistency with CBL iOS.

**Comment by hideki** [comment]
Hi @mtrakal,
Both `PasswordAuthorizer` and `BasicAuthenticator` are internal class. Please use folllowing method to create `Authenticator` for basic authentication.
```java
Authenticator basicAuth = AuthenticatorFactory.createBasicAuthenticator("username", "password");
```

We know our API is messy.  We could not isolate well between `Couchbase public API` and `Java public access level modifier`. We will fix this with v2.0 release.
